framework/src/base/Assembly.C

```diff
@@ -1846,8 +1879,7 @@ Assembly::reinit(const Elem * elem, unsigned int side, const std::vector<Point>
   _current_qrule_arbitrary_face = qrules(elem_dimension).arbitrary_face.get();
 
   // Make sure the qrule is the right one
-  if (_current_qrule_face != _current_qrule_arbitrary_face)
-    setFaceQRule(_current_qrule_arbitrary_face, elem_dimension);
+  setFaceQRule(_current_qrule_arbitrary_face, elem_dimension);
```

Comment:
why is if statement removed?

Reply:
I had been experimenting with moving that if check down into the setFaceQRule function, but it turns out there are apparently side effects of calling that function that cause other things to break if we _always_ perform that check instead of just whenever the caller wants to :-(  This was just an oversight after that failed attempt - I'll put this if check back in.